test/functional/feature_rbf.py

```diff
@@ -550,86 +550,80 @@ def test_prioritised_transactions(self):
+            sequence=BIP125_SEQUENCE_NUMBER,
+            fee_rate=Decimal('0.01'),
+        )
+        assert_equal(True, self.nodes[0].getmempoolentry(optin_parent_tx['txid'])['bip125-replaceable'])
+
+        replacement_parent_tx = wallet.create_self_transfer(
+            from_node=self.nodes[0],
+            utxo_to_spend=confirmed_utxo,
+            sequence=BIP125_SEQUENCE_NUMBER,
+            fee_rate=Decimal('0.02'),
+        )
 
         # Test if parent tx can be replaced.
-        res = self.nodes[0].testmempoolaccept(rawtxs=[replacement_parent_tx['hex']], maxfeerate=0)[0]
+        res = self.nodes[0].testmempoolaccept(rawtxs=[replacement_parent_tx['hex']])[0]
 
         # Parent can be replaced.
         assert_equal(res['allowed'], True)
 
         # Create an opt-out child tx spending the opt-in parent
-        optout_child_tx = self.nodes[0].createrawtransaction([{
-            'txid': optin_parent_txid,
-            'vout': 0,
-            "sequence": 0xffffffff,
-        }], {self.nodes[0].getnewaddress(): Decimal("9.99990")})
-
-        optout_child_tx = self.nodes[0].signrawtransactionwithwallet(optout_child_tx)
-
-        # Broadcast child tx
-        optout_child_txid = self.nodes[0].sendrawtransaction(hexstring=optout_child_tx["hex"], maxfeerate=0)
-        assert optout_child_txid in self.nodes[0].getrawmempool()
-
-        replacement_child_tx = self.nodes[0].createrawtransaction([{
-            'txid': optin_parent_txid,
-            'vout': 0,
-            "sequence": 0xffffffff,
-        }], {self.nodes[0].getnewaddress(): Decimal("9.00000")})
-
-        replacement_child_tx = self.nodes[0].signrawtransactionwithwallet(replacement_child_tx)
+        parent_utxo = wallet.get_utxo(txid=optin_parent_tx['txid'])
+        optout_child_tx = wallet.send_self_transfer(
+            from_node=self.nodes[0],
+            utxo_to_spend=parent_utxo,
+            sequence=0xffffffff,
+            fee_rate=Decimal('0.01'),
+        )
+
+        # Reports true due to inheritance
+        assert_equal(True, self.nodes[0].getmempoolentry(optout_child_tx['txid'])['bip125-replaceable'])
+
+        replacement_child_tx = wallet.create_self_transfer(
+            from_node=self.nodes[0],
+            utxo_to_spend=parent_utxo,
+            sequence=0xffffffff,
+            fee_rate=Decimal('0.02'),
+            mempool_valid=False,
+        )
 
         # Broadcast replacement child tx
         # BIP 125 :
         # 1. The original transactions signal replaceability explicitly or through inheritance as described in the above
         # Summary section.
-        # The original transaction (`optout_child_tx`) doesn't signal RBF but its parent (`optin_parent_txid`) does.
+        # The original transaction (`optout_child_tx`) doesn't signal RBF but its parent (`optin_parent_tx`) does.
         # The replacement transaction (`replacement_child_tx`) should be able to replace the original transaction.
```

Comment:
Is my understanding here correct?

a) `testmempoolaccept` returns that `replacement_child_tx` is allowed even though it is rejected w/ 'txn-mempool-conflict'
b) `getmempoolentry` returns that `optout_child_tx` is 'bip125-replaceable' - as you descbribe in https://github.com/bitcoin/bitcoin/issues/22209 even though it is not

So does anything need to be done to fix a and b?

Another question I have (which maybe I should just try out). But, I wonder what would happen if you tried something like https://github.com/bitcoin/bitcoin/pull/22210/files#diff-260c479178e3e99a16798aeed39f400676894392864784eda6c207716704398dR584-R589 after `optout_child_tx` was already sent

Reply:
a) No, it is also rejected by testmempoolaccept, see `mempool_valid=False` for `replacement_child_tx` above.
b) correct

> So does anything need to be done to fix a and b?

Maybe. But a fix should be done outside a test-only change.